src/text-field.js

```diff
@@ -30,81 +27,48 @@ class Textfield extends React.Component {
   }
 
   render(){
-
     const {
-      className,
-      type,
-      label,
-      floatingLabel,
-      error,
+      children,
       expandable,
-      button,
       icon,
-      expandableHolder
+      floating,
+      className
     } = this.props;
 
-    let {
-      id
-    } = this.props;
-
-    const outerClasses = classnames(outerBaseClasses, {
-      'mdl-textfield--floating-label': floatingLabel,
-      'mdl-textfield--expandable': expandable,
-      'mdl-button': button,
-      'mdl-js-button': button,
-      'mdl-button--icon': icon,
-      'mdl-textfield__expandable-holder': expandableHolder
-    }, className);
-
-    const inputClasses = classnames(inputBaseClasses);
+    let {id} = this.props;
 
-    const labelClasses = classnames(labelBaseClasses);
-
-    let textField;
-    let labelField;
-    let errorField;
-
-    if (!id) {
-      id = '_' + Math.random().toString(36).slice(2);
+    if(!id){
+      id = this._autoId;
```

Comment:
This should be removed, the `autoId` is only used for cloning and isn't passed down from props.